zap/src/parser/reports.rs

```diff
@@ -273,7 +290,13 @@ impl<'src> Report<'src> {
 				format!("(inclusive) max: {}", max),
 			]),
 			Self::AnalyzeInvalidOptionalType { .. } => Some(vec![
-				"this type can only be used when it is not marked as optional".to_string(),
+				"you cannot have 'double optional' types, where a type is optional twice".to_string(),
+				"maps cannot have optional keys or values, as those are impossible to represent in luau".to_string(),
+				"additionally the `unknown` type cannot be optional".to_string(),
```

Comment:
again we should use the props to show the 'correct' one here

Reply:
also out of scope for this PR